lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1673,6 +1674,52 @@ bool TypeChecker::typeCheckFunctionBodyUntil(FuncDecl *FD,
   BraceStmt *BS = FD->getBody();
   assert(BS && "Should have a body");
 
+  // FIXME: FIXME_NOW: DETERMINE: What is the appropriate place for this code to
+  //        live?
+  if (FD->hasSingleExpressionBody()) {
+    auto fnTypeLoc = FD->getBodyResultTypeLoc();
+    auto fnType = fnTypeLoc.getType();
+    auto returnStmt = cast<ReturnStmt>(BS->getElement(0).get<Stmt *>());
+    auto E = returnStmt->getResult();
```

Comment:
Nitpick: shouldn't this use the accessor for `hasSingleExpressionBody`?

Reply:
Yup!  Fixed in 
https://github.com/apple/swift/pull/23251/commits/a310279dabda30f1de50b2a6bb2c11c0ef20868f .